descartes_light/solvers/include/descartes_light/solvers/bgl/impl/bgl_ladder_graph_solver.hpp

```diff
@@ -0,0 +1,281 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2021, Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <descartes_light/descartes_macros.h>
+DESCARTES_IGNORE_WARNINGS_PUSH
+#include <boost/graph/dijkstra_shortest_paths.hpp>
+#include <console_bridge/console.h>
+#include <omp.h>
+DESCARTES_IGNORE_WARNINGS_POP
+
+#include <descartes_light/solvers/bgl/bgl_ladder_graph_solver.h>
+#include <descartes_light/types.h>
+
+#define UNUSED(x) (void)(x)
+
+static void reportFailedEdges(const std::vector<std::size_t>& indices)
+{
+  if (indices.empty())
+    CONSOLE_BRIDGE_logInform("No failed edges");
+  else
+  {
+    std::stringstream ss;
+    ss << "Failed edges:\n";
+    for (const auto& i : indices)
+      ss << "\t" << i << "\n";
+
+    CONSOLE_BRIDGE_logWarn(ss.str().c_str());
+  }
+}
+
+static void reportFailedVertices(const std::vector<std::size_t>& indices)
+{
+  if (indices.empty())
+    CONSOLE_BRIDGE_logInform("No failed vertices");
+  else
+  {
+    std::stringstream ss;
+    ss << "Failed vertices:\n";
+    for (const auto& i : indices)
+      ss << "\t" << i << "\n";
+
+    CONSOLE_BRIDGE_logWarn(ss.str().c_str());
+  }
+}
+
+namespace descartes_light
+{
+template <typename FloatType>
+BGLLadderGraphSolver<FloatType>::BGLLadderGraphSolver(const std::size_t dof, unsigned num_threads)
+  : dof_{ dof }, num_threads_{ num_threads } {};
+
+template <typename FloatType>
+BuildStatus BGLLadderGraphSolver<FloatType>::buildImpl(
+    const std::vector<typename WaypointSampler<FloatType>::ConstPtr>& trajectory,
+    const std::vector<typename EdgeEvaluator<FloatType>::ConstPtr>& edge_evaluators,
+    const std::vector<typename StateEvaluator<FloatType>::ConstPtr>& state_evaluators)
+{
+  BuildStatus status;
+
+  // Build Vertices
+  ladder_rungs_.resize(trajectory.size());
+  long cnt = 0;
+
+  using Clock = std::chrono::high_resolution_clock;
+  std::chrono::time_point<Clock> start_time = Clock::now();
+#pragma omp parallel for num_threads(num_threads_)
+  for (std::size_t i = 0; i < trajectory.size(); ++i)
```

Comment:
This must be an integral type explained [here](https://stackoverflow.com/questions/2820621/why-arent-unsigned-openmp-index-variables-allowed). Recommend changing to long.